pkg/flow/internal/controller/node_config_import.go

```
@@ -0,0 +1,328 @@
+	cn.processImportedContent(parsedImportedContent)
+
+	// evaluate the importConfigNodesChildren that have been created
+	err = cn.evaluateChildren()
+	if err != nil {
+		level.Error(cn.logger).Log("msg", "failed to evaluate nested import", "err", err)
+		return
+	}
+
+	// trigger to stop previous children from running and to start running the new ones.
+	if cn.importChildrenRunning {
+		cn.importChildrenUpdateChan <- struct{}{}
+	}
+
+	cn.OnBlockNodeUpdate(cn)
+}
+
+// processImportedContent processes declare and import blocks of the provided ast content.
+func (cn *ImportConfigNode) processImportedContent(content *ast.File) {
+	for _, stmt := range content.Body {
+		switch stmt := stmt.(type) {
+		case *ast.BlockStmt:
+			fullName := strings.Join(stmt.Name, ".")
+			switch fullName {
+			case "declare":
+				cn.processDeclareBlock(stmt)
+			case importsource.BlockImportFile, importsource.BlockImportString: // TODO: add other import sources
+				cn.processImportBlock(stmt, fullName)
+			default:
+				level.Error(cn.logger).Log("msg", "only declare and import blocks are allowed in a module", "forbidden", fullName)
+			}
+		default:
+			level.Error(cn.logger).Log("msg", "only declare and import blocks are allowed in a module")
+		}
+	}
+}
+
+// processDeclareBlock stores the declare definition in the importedDeclares.
+func (cn *ImportConfigNode) processDeclareBlock(stmt *ast.BlockStmt) {
+	if _, ok := cn.importedDeclares[stmt.Label]; ok {
+		level.Error(cn.logger).Log("msg", "declare block redefined", "name", stmt.Label)
+		return
+	}
+	cn.importedDeclares[stmt.Label] = stmt.Body
+}
+
+// processDeclareBlock creates an ImportConfigNode child from the provided import block.
+func (cn *ImportConfigNode) processImportBlock(stmt *ast.BlockStmt, fullName string) {
+	sourceType := importsource.GetSourceType(fullName)
+	if _, ok := cn.importConfigNodesChildren[stmt.Label]; ok {
+		level.Error(cn.logger).Log("msg", "import block redefined", "name", stmt.Label)
+		return
```

Should this error be raised up higher? Should this cause something else to fail rather than being constrained to logs?